### PR TITLE
Added static method for DecimalTextIntParser

### PR DESCRIPTION
**Pull request by gruay**
I added the static method for DecimalTextIntParser (issue #427) the same way is done in DecimalTextLongParser.
I'm planning on adding static methods for the rest of the parsers.

### CODE REVIEW

**Comment by rmetzger** [comment]
Hey,
thank you for your pull request.
Is there a way to use only one method for the parsing? I have the impression that `public int parseField()` and `public static final int parseField()` are quite similar.

**Comment by gruay** [comment]
Right now the non static one is returning the position in the byte array where it ended parsing the int (or -1 if something went wrong) and then the parsed number is put in a parameter given, whereas the static one simply returns the parsed int.
So, they both do the same but the non-static one gives more information.
I'm not sure if that answer your question now that I re-read it

**Comment by rmetzger** [comment]
I was just wondering if there would be a way to avoid duplicate code, for example by introducing an "internal" method that can handle both cases?

**Comment by gruay** [comment]
Sure, I could move the non-static code to a new static method that would do the same. Then the non-static method would call that one for getting both the ending position and the value and the static method would just care about the parsed value.
Do you mean something like this?

**Comment by gruay** [comment]
Hey, I just created the new static method that is called from both methods.
Did you mean something like this?

File: stratosphere-core/src/main/java/eu/stratosphere/types/parser/DecimalTextIntParser.java [hunk 1]
```diff
@@ -28,19 +28,40 @@
...
+			int end = parseIntField(bytes, startPos, limit, delimiter, reusable);
+			this.result = reusable;
+			return end;
+		} 
+		catch (NumberFormatException e) {
+			if (e.getMessage() == null) {
+				setErrorState(ParseErrorState.NUMERIC_VALUE_ILLEGAL_CHARACTER);
+				return -1;
+			}
+			else if (e.getMessage().equals("Orphaned minus sign.")) {
```
Comment (rmetzger) [inline stratosphere-core/src/main/java/eu/stratosphere/types/parser/DecimalTextIntParser.java:41]: Differentiating between exceptions by their messages is very bad design.
Imagine someone changes the exception messages. (You could use constants for that)
BUT: a) you could throw different kinds of exceptions.
b) you could implement a custom Exception class with an Enum to represent the different error states.

**Comment by StephanEwen** [comment]
I agree with Robert. That is very fragile code, the way the error messages and return codes are handled. Also, throwing an exception and then turning that into a state seems reversed. Why not parse and set the state, and then react with an exception to the state (if it is not the "OKAY" state)?

**Comment by StephanEwen** [comment]
I am not sure about the use case of this method, but I have a feeling that the delimiter is odd. I think @sscdotopen 's initial case was that it is odd to come up with some delimiter only to have a standalone array parsed.

The original parse method is used to parse a field as part of a CSV row. The standalone method ,seems to me, it rather intended to parse an isolated value inside an array.

@sscdotopen Can you comment on this?

**Comment by StephanEwen** [comment]
What do we do with this pull request? I don't want to merge it in the state that it is in, I think it needs to be improved.

Is anything going to happen here?

**Comment by fhueske** [comment]
I think the problem is, that it is not really clear what issue #427 is about...

**Comment by StephanEwen** [comment]
I think this issue simply asks for a utility to parse binary text data (as
read from a file stream) directly into a primitive type.

Is it fair to ask whoever really needs it should implement it? I think it
is not a priority thing right now.

**Comment by StephanEwen** [comment]
I will close this pull request, because the code is not in shape to be merged and the functionality has by now been subsumed by a1555ce84ae47edb8cd696f672935cbeffd3481b including tests.

**Pull request closed without merging**